sdk_extensions/aws_v1_support/src/main/java/io/opentelemetry/sdk/extensions/trace/aws/resource/EcsResource.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.sdk.extensions.trace.aws.resource;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Strings;
+import io.opentelemetry.common.Attributes;
+import io.opentelemetry.sdk.resources.ResourceConstants;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+class EcsResource extends AwsResource {
+
+  private static final Logger logger = Logger.getLogger(EcsResource.class.getName());
+
+  private static final String ECS_METADATA_KEY_V4 = "ECS_CONTAINER_METADATA_URI_V4";
+  private static final String ECS_METADATA_KEY_V3 = "ECS_CONTAINER_METADATA_URI";
+
+  private final Map<String, String> sysEnv;
+
+  EcsResource() {
+    this(System.getenv());
+  }
+
+  @VisibleForTesting
+  EcsResource(Map<String, String> sysEnv) {
+    this.sysEnv = sysEnv;
+  }
+
+  @Override
+  Attributes createAttributes() {
+    if (!isOnEcs()) {
+      return Attributes.empty();
+    }
+
+    Attributes.Builder attrBuilders = Attributes.newBuilder();
+    try {
+      String hostName = InetAddress.getLocalHost().getHostName();
+      attrBuilders.setAttribute(ResourceConstants.CONTAINER_NAME, hostName);
+
+    } catch (UnknownHostException e) {
+      logger.log(Level.WARNING, "Could not get docker container name from hostname.", e);
+    }
+
+    return attrBuilders.build();
+  }
+
+  private boolean isOnEcs() {
+    if (sysEnv == null) {
```

Comment:
Avoid unnecessary null checks, `sysEnv` can't be null (unless a unit test passes it randomly which is something we don't need to worry about in business logic)

Reply:
removed null check and synEnv null test